### PR TITLE
MM-48186: Add a new API endpoint to add a user to their default GroupChannels and GroupTeams.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary

* Adds a new API endpoint (`POST /api/v4/ldap/users/:user_id/group_sync_memberships`) to add a single user to all of their group-channels and group-teams.
* Changes the signature of `(app.AppIface).CreateDefaultMemberships` to accept a struct to reduce the number of method parameters.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-48186

#### Release Note

```release-note
Added a new API endpoint `POST /api/v4/ldap/users/:user_id/group_sync_memberships` to add (or re-add) a user to all of their default teams and channels for all of the groups they're a member of.
```

#### Related

* https://github.com/mattermost/enterprise/pull/1291
* https://github.com/mattermost/mattermost-webapp/pull/11536

### CODE REVIEW

File: app/channels.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func NewChannels(s *Server, services map[ServiceKey]any) (*Channels, error) {
 	for _, svcKey := range requiredServices {
 		svc, ok := services[svcKey]
 		if !ok {
-			return nil, fmt.Errorf("Service %s not passed", svcKey)
+			return nil, fmt.Errorf("service %s not passed", svcKey)
```
Comment (mkraft) [inline app/channels.go:123]: unrelated lint fix

File: api4/ldap.go [hunk 2]
```diff
@@ -419,3 +420,25 @@ func removeLdapPrivateCertificate(c *Context, w http.ResponseWriter, r *http.Req
...
+func addUserToGroupSyncables(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleWriteUserManagementGroups) {
+		c.SetPermissionError(model.PermissionSysconsoleWriteUserManagementGroups)
+		return
+	}
+
+	auditRec := c.MakeAuditRecord("addUserToGroupSyncables", audit.Fail)
+	defer c.LogAuditRec(auditRec)
+
+	params := model.CreateDefaultMembershipParams{Since: 0, ReAddRemovedMembers: true, ScopedUserID: &c.Params.UserId}
```
Comment (mkraft) [inline api4/ldap.go:435]: pondering if we need to validate the user is LDAP...

File: i18n/en.json [hunk 3]
```diff
@@ -147,6 +147,10 @@
     "id": "api.admin.saml.set_certificate_from_metadata.missing_content_type.app_error",
     "translation": "Missing content type."
   },
+  {
+    "id": "api.admin.syncables_error",
+    "translation": " "
```
Comment (mkraft) [inline i18n/en.json:152]: purposely empty translation that is not necessary in the UI

File: model/group.go [hunk 4]
```diff
@@ -247,3 +247,11 @@ type GroupsWithCount struct {
 	Groups     []*Group `json:"groups"`
 	TotalCount int64    `json:"total_count"`
 }
+
+type CreateDefaultMembershipParams struct {
+	Since               int64
+	ReAddRemovedMembers bool
+	ScopedUserID        *string
+	ScopedTeamID        *string
+	ScopedChannelID     *string
```
Comment (mkraft) [inline model/group.go:256]: We could very easily have this accept a slice of user IDs, but it's not currently in the requirement.

File: model/group.go [hunk 5]
```diff
@@ -247,3 +247,11 @@ type GroupsWithCount struct {
 	Groups     []*Group `json:"groups"`
 	TotalCount int64    `json:"total_count"`
 }
+
+type CreateDefaultMembershipParams struct {
+	Since               int64
+	ReAddRemovedMembers bool
+	ScopedUserID        *string
```
Comment (mkraft) [inline model/group.go:254]: We could easily accept a slice of user IDs but it's not currently in the requirement.

**Review by plant99 — APPROVED** [review]

**Review by plant99 — APPROVED** [review]

**Review by BenCookie95 — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Working fine
